custom_components/grocy/binary_sensor.py

```diff
@@ -13,41 +14,60 @@
 ):  # pylint: disable=unused-argument
     """Setup binary_sensor platform."""
     async_add_entities(
-        [GrocyExpiringProductsBinarySensor(hass, discovery_info)], True)
+        [GrocyBinarySensor(hass, discovery_info)], True)
 
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Setup sensor platform."""
+    for binary_sensor in BINARY_SENSOR_TYPES:
+        async_add_devices([GrocyBinarySensor(hass, binary_sensor)], True)
 
-class GrocyExpiringProductsBinarySensor(BinarySensorDevice):
+class GrocyBinarySensor(BinarySensorDevice):
     """grocy binary_sensor class."""
 
-    def __init__(self, hass, config):
+    def __init__(self, hass, sensor_type):
         self.hass = hass
+        self.sensor_type = sensor_type
         self.attr = {}
         self._status = False
-        self._name = config.get("name", DEFAULT_NAME) + ".expiring_products"
+        self._name = '{}.{}'.format(DEFAULT_NAME, self.sensor_type)
         self._client = self.hass.data[DOMAIN_DATA]["client"]
 
     async def async_update(self):
         import jsonpickle
         """Update the binary_sensor."""
         # Send update "signal" to the component
-        await self._client.async_update_expiring_products()
+        await self._client.async_update_data(self.sensor_type)
 
         # Get new data (if any)
-        expiring_products = (
-            self.hass.data[DOMAIN_DATA].get("expiring_products"))
+        data = self.hass.data[DOMAIN_DATA].get(self.sensor_type)
 
         # Check the data and update the value.
-        if not expiring_products:
+        if not data:
             self._status = self._status
         else:
             self._status = True
 
         # Set/update attributes
         self.attr["attribution"] = ATTRIBUTION
         self.attr["items"] = jsonpickle.encode(
-            expiring_products,
+            data,
             unpicklable=False)
 
+    @property
+    def unique_id(self):
+        """Return a unique ID to use for this binary_sensor."""
+        return (
+            "e308a906-4c3f-4a58-a067-47d04d3e196e" + self.name
```

Comment:
is this the right way to specify an unique_id?

Reply:
I'd be hesistant to use a generated uuid for our sensors, not sure of grocy provides some kind of id. Otherwise we have to use various parameters and maybe hash them or something to generate a uuid. Maybe we could also replace the uuid with an (md5?) hash of the api key. This would ensure stable identifiers as long the api key doesn'T change.

Reply:
https://developers.home-assistant.io/docs/en/entity_registry_index.html

Reply:
I will try your method and show result.
If we have duplicate sensors (*_2) : not a good id.
Keep in touch :)